Src/ILGPU/Runtime/PeerAccess.cs

```diff
@@ -35,39 +35,43 @@ partial class Accelerator
         /// </summary>
         private void PeerAccessAcceleratorDestroyed(object sender, EventArgs e)
         {
-            if (sender is Accelerator otherAccelerator)
-                DisablePeerAccess(otherAccelerator);
+            // Reject cases in which the sender is not another accelerator instance
+            if (!(sender is Accelerator otherAccelerator))
+                return;
+            // Disable the peer access on this one
+            otherAccelerator.DisablePeerAccess(this);
```

Comment:
QUESTION: Should we disable peer access the other way?
i.e `this` accelerator should not be accessing the just destroyed `otherAccelerator`.

Reply:
This event will be invoked prior to actual destroying it. That's why I think it "should work". Alternatively, we can try to change the "direction" of the internal event processing. Do you have a different solution in mind?

Reply:
The event is raised just before `otherAccelerator` is destroyed. And the call `otherAccelerator.DisablePeerAccess(this);` removes access from `otherAccelerator` to `this` accelerator.

If `this` accelerator also has peer access to `otherAccelerator` (i.e. bidirectional peer access), should we also call `this.DisablePeerAccess(otherAccelerator);`?

Reply:
Sorry, I misunderstood your comment at first. I think this makes a lot of sense, since multiple calls to `DisablePeerAccess` should not fail anyway.